src/assets/all_assets_request.rs

```diff
@@ -118,9 +90,15 @@ impl AllAssetsRequest {
     /// # Returns
     /// The request object
     /// [AllAssetsRequest](struct.AllAssetsRequest.html)
-    pub fn set_asset_issuer(&mut self, asset_issuer: &str) -> &mut Self {
-        self.asset_issuer = Some(asset_issuer.to_owned());
-        self
+    pub fn set_asset_issuer(self, asset_issuer: &str) -> Result<AllAssetsRequest, String> {
+        if asset_issuer.len() != 56 {
+            return Err("asset_issuer must be 56 characters".to_string());
+        }
```

Comment:
These are still runtime errors. We could introduce an 'InvalidAssetRequest' state and return this if the asset_issuer > 12.

Reply:
moved comment to https://github.com/Baseflow/stellar-rust-sdk/issues/8